### PR TITLE
Fix hanging api call, parsing errors table bug, and update tests

### PR DESCRIPTION
**Pull request by cwcrystal8**
1. Fixed a bug where the parsing errors table would appear even though there was no error. The problem with the code was it was missing a check for errors before adding them to the list.

2. Fixed a bug where the node page would fail to render properly. The problem was a missing check for non-null labels before iterating through them.
Switch over to new API

3. Updated the test cases and source code for changes in the behavior of the API. 

4. Reran build and updated static files to reflect the new src changes.

### CODE REVIEW

File: mcf-browser/src/back-end/server-api.js [hunk 1]
```diff
@@ -54,11 +54,15 @@ async function readFileList(fileList) {
             const mcfParser = new ParseMcf(tmcf.name + '&' + file.name);
             return mcfParser.parseMcfStr(mcf);
           });
-        finalReturn['errMsgs'] =
+
+        if(tmcfOut['errMsgs'] && tmcfOut['errMsgs'].length !== 0) {
+
+          finalReturn['errMsgs'] =
           finalReturn['errMsgs'].concat({
```
Comment (pradh) [inline mcf-browser/src/back-end/server-api.js:61]: The indents seem off, could you run lint?

File: mcf-browser/src/back-end/utils.js [hunk 2]
```diff
@@ -106,7 +106,7 @@ async function doesExistsInKG(dcid) {
   const url = API_ROOT + '/node/triples?dcids=' + dcid + '&limit=1';
   return fetch(url)
       .then((res) => res.json())
-      .then((data) => JSON.parse(data.payload)[dcid] ? true : false);
+      .then((data) => (JSON.parse(data.payload)[dcid] && JSON.parse(data.payload)[dcid].length > 0) ? true : false);
```
Comment (pradh) [inline mcf-browser/src/back-end/utils.js:109]: Perhaps this could use a comment?

File: mcf-browser/src/back-end/server-api.js [hunk 3]
```diff
@@ -54,11 +54,15 @@ async function readFileList(fileList) {
             const mcfParser = new ParseMcf(tmcf.name + '&' + file.name);
             return mcfParser.parseMcfStr(mcf);
           });
-        finalReturn['errMsgs'] =
+
+        if(tmcfOut['errMsgs'] && tmcfOut['errMsgs'].length !== 0) {
```
Comment (pradh) [inline mcf-browser/src/back-end/server-api.js:58]: Just wondering why just having the second condition may not work?  Given https://github.com/datacommonsorg/tools/blob/4bd98c15a4fd309ece98301da4946234da24b61a/mcf-browser/src/back-end/parse-mcf.js#L69

**Review by pradh — COMMENTED** [review]
Nice first PR!

Does this PR actually include the switch to new API as the description suggests or did I miss it?

File: mcf-browser/src/back-end/utils.js (hunk 2 above)
Comment (cwcrystal8) [reply mcf-browser/src/back-end/utils.js:109]: Added one!

File: mcf-browser/src/back-end/server-api.js (hunk 1 above)
Comment (cwcrystal8) [reply mcf-browser/src/back-end/server-api.js:61]: Is there a specific linter we use in DC?

**Comment by cwcrystal8** [comment]
ptal, the code has the following changes:

- Fixed indentation
- Removed unnecessary condition in parsing error if statement
- Added a comment explaining the logic behind checking for DCIDs

Also, @pradh this PR does not include the switch to the new API, it just fixes the back-end issue in server-api.js. New API switch will be in the next PR, sorry for the confusion!

File: mcf-browser/src/back-end/server-api.js (hunk 1 above)
Comment (pradh) [reply mcf-browser/src/back-end/server-api.js]: Website uses this - https://github.com/datacommonsorg/website/blob/master/run_test.sh#L49-L52

**Comment by pradh** [comment]
> 

Ok, could update the description then: https://github.com/datacommonsorg/website/blob/master/run_test.sh#L49-L52

**Review by pradh — DISMISSED** [review]
Thanks for the changes!

**Comment by cwcrystal8** [comment]
Updated the description! I just realized you were referring to point (3), which was confusingly worded. Sometime between the last update and this PR, the behavior of the current (old) API changed so I updated the test cases to reflect that new behavior. I reworded point (3) to make this clearer.

File: mcf-browser/asset-manifest.json [hunk 4]
```diff
@@ -1,22 +1,23 @@
 {
```
[the remark this answers is not in the document]
Comment (cwcrystal8) [reply mcf-browser/asset-manifest.json:1]: Yep, the file was automatically generated using `npm run build`

File: mcf-browser/src/back-end/server-api.js [hunk 5]
```diff
@@ -54,11 +54,15 @@ async function readFileList(fileList) {
...
             return mcfParser.parseMcfStr(mcf);
           });
-        finalReturn['errMsgs'] =
-          finalReturn['errMsgs'].concat({
-            'file':tmcf.name,
-            'errs': tmcfOut['errMsgs'],
-          });
+
+        if(tmcfOut['errMsgs'].length !== 0) {
+
```
[the remark this answers is not in the document]
Comment (cwcrystal8) [reply mcf-browser/src/back-end/server-api.js:59]: fixed in next commit

File: mcf-browser/src/back-end/server-api.js [hunk 6]
```diff
@@ -54,11 +54,15 @@ async function readFileList(fileList) {
             const mcfParser = new ParseMcf(tmcf.name + '&' + file.name);
             return mcfParser.parseMcfStr(mcf);
           });
-        finalReturn['errMsgs'] =
-          finalReturn['errMsgs'].concat({
-            'file':tmcf.name,
-            'errs': tmcfOut['errMsgs'],
-          });
+
+        if(tmcfOut['errMsgs'].length !== 0) {
```
[the remark this answers is not in the document]
Comment (cwcrystal8) [reply mcf-browser/src/back-end/server-api.js:58]: fixed in next commit

File: mcf-browser/src/back-end/graph.js [hunk 7]
```diff
@@ -178,6 +178,10 @@ class Node {
    *     the calling Node is the source of the triple.
    */
   async createAssertionsFromLabels(propLabels, isInverse) {
+    if(!propLabels || propLabels.length === 0){
```
[the remark this answers is not in the document]
Comment (cwcrystal8) [reply mcf-browser/src/back-end/graph.js:181]: fixed in next commit

**Comment by cwcrystal8** [comment]
@beets ptal, the whitespace issues should be fixed now

**Review by beets — APPROVED** [review]

**Pull request merged**